src/main/java/com/carpool/car_pool/services/RideOfferService.java

```diff
@@ -91,4 +90,15 @@ public RideOfferResponse editRideOfferDetail(EditRideOfferRequest editRideofferR
 
 
     }
+
+    public void deleteRideOffer(long id, UserEntity user) {
+        RideOfferEntity rideOffer = rideOfferRepository.findById(id)
+                .orElseThrow(()-> new RuntimeException("Ride offer does not exist"));
+
+        if (!rideOffer.getCreator().equals(user)){
+            throw new RuntimeException("Only owner can delete this ride offer");
+        }
+
+        rideOfferRepository.delete(rideOffer);
```

Comment:
Entity doesn't have orphan removal so requests for deleted ride offer will still exist. You can delete them manually here or make changes to entity good example is in UserEntity in fields that have OneToMany annotation